rfcs/2021-07-23-8447-extract-helm-charts.md

```diff
@@ -0,0 +1,119 @@
+tag version), thus any usage of Helm's testing CLI `ct` would involve a custom configuration that would need to be updated for
+each release (and duplicated for any test config).
+
+Today our `test-e2e-kubernetes` suite does double duty (arguably triple) by both running smoke/integration tests and verifying
+that our charts are designed and configured properly. This isn't efficient as we need to compile Vector and build an image to
+test any changes to our charts which turns a process that could be a couple minutes into half an hour ([based on our CI dashboard](https://app.datadoghq.com/metric/explorer?from_ts=1627251689494&to_ts=1627445043208&live=false&tile_size=l&exp_metric=gh.actions.workflow_job.execution_secs.99percentile&exp_scope=conclusion%3Asuccess%2Cworkflow%3Ak8s_e2e_suite&exp_group=workflow&exp_agg=max&exp_row_type=metric#workflow:test_suite)).
+
+The final issue is that today we have coupled chart releases directly to Vector releases, we publish chart versions in lockstep
+which causes either delays in chart improvements (or fixes) or unnecessary Vector releases to address chart updates.
+
+## User Experience
+
+Moving our charts into the `DataDog/helm-charts` repository would require users to add Datadog's Helm repository address
+(`https://helm.datadoghq.com`) to use our charts. This would not be breaking for charts installed from the old `timberio` repository,
+they would be able to seamlessly upgrade to new charts from the new Helm repository. As an example, the [upgrade instructions for Helm](https://vector.dev/docs/setup/installation/package-managers/helm/#management)
+would be changed like so (merely changing the repository name from `timberio` to `datadog`):
+
+```shell
+helm repo update && helm upgrade --namespace vector vector datadog/vector-agent --reuse-values
+```
+
+Users that have installed charts through cloning our repository and installing via their filesystem would need to clone and track
+the new repository, but this method is not documented, or advised.
+
+We can maintain backward compatibility for users referencing our raw or kustomize resources.
+
+## Implementation
+
+The extraction is straightforward. Migrate the contents of `distribution/helm` into the existing `DataDog/helm-charts` git repository.
+
+Existing CI jobs for our charts can be transfered to the new repository with minor changes, or replaced with the tooling available
+with `ct`. Our existing lint job calls a make target that lints all charts in the `distribution/helm` directory and could be replaced
+with the `ct lint` command, which is essentially the same but contains additional helpers.
+
+We can leverage DataDog's [script](https://github.com/DataDog/datadog-agent/blob/main/Dockerfiles/manifests/generate.sh) or update our
+existing scripts to keep the raw and kustomize resources in sync with changes to the `timberio/helm-charts` repository. This could be
```

Comment:
These seem like details we can figure out later, but for the purposes of RFCs we typically prefer to pick a specific path.

Reply:
Fair, the differences are fairly slight and I'd implement whichever was least effort. I can do the research ahead of time if need be.